src/Tools/dotnet-monitor/EndpointInfo/ClientEndpointInfoSource.cs

```diff
@@ -15,8 +15,20 @@ namespace Microsoft.Diagnostics.Tools.Monitor
 {
     internal sealed class ClientEndpointInfoSource : IEndpointInfoSourceInternal
     {
+        // The amount of time to wait before abandoning the attempt to create an EndpointInfo from
+        // the enumerated processes. This may happen if a runtime instance is unresponsive to
+        // diagnostic pipe commands.
+        private readonly TimeSpan AbandonProcessTimeout = TimeSpan.FromSeconds(1);
```

Comment:
Can you try this out with a very large number of processes? Will 1 second be enough to schedule and finish all the associated tasks?

Reply:
I can make the timeout more generous, something like 3 or 5 seconds. We don't want to make it too long because then a single unresponsive process will cause all HTTP requests to be delayed by the timeout period. Maybe I can add a warning logging statement when abandoning getting process information for processes to make it more obvious to the user why requests take a "long" time.